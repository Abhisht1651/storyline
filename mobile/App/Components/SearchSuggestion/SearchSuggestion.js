// @flow

import React, { useState } from 'react'
import { View, Text, TextInput, ScrollView, Platform } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import styles from '../Styles/SearchSuggestionStyles/SearchSuggestionStyle'

type Props = {
  placeholder: string,
  rightEndText: string,
  zIndex: number
}

const SearchSuggestion = (props: Props) => {
  const [value, setValue] = useState(false)
  const [searchItem, setSearchItem] = useState('')

  const { placeholder, rightEndText, zIndex } = props

  const changeTextHandler = (text: string) => {
    let val = false
    if (text !== '') {
      val = true
    }
    setValue(val)
    setSearchItem(text)
  }

  const closeDropDown = () => {
    setValue(false)
  }

  return (
    <View style={styles.container}>
      <View style={styles.searchBar}>
        <TextInput
          placeholder={placeholder}
          style={styles.inputField}
          selectionColor="#7B46E4"
          keyboardType={Platform.OS === 'ios' ? 'default' : 'visible-password'}
          onChangeText={changeTextHandler}
          onBlur={closeDropDown}
        />
        <Text style={styles.optionalText}>{rightEndText}</Text>
      </View>

      {value && (
        <View style={styles.dropDownBody} zIndex={zIndex}>
          <ScrollView>
            <TouchableOpacity>
              <View style={styles.listItem}>
                <Text style={styles.listItemText}>{searchItem}</Text>
                <Text style={styles.randomText}>randomtext1</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity>
              <View style={styles.listItem}>
                <Text style={styles.listItemText}>{searchItem}</Text>
                <Text style={styles.randomText}>randomtext2</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity>
              <View style={styles.listItem}>
                <Text style={styles.listItemText}>{searchItem}</Text>
                <Text style={styles.randomText}>randomtext3</Text>
              </View>
            </TouchableOpacity>
          </ScrollView>
        </View>
      )}
    </View>
  )
}

SearchSuggestion.defaultProps = {
  placeholder: 'Search',
  rightEndText: '(Optional)',
  zIndex: 1
}

export default SearchSuggestion
