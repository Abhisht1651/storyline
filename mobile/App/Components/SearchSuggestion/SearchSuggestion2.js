// @flow

import React, { useState } from 'react'
import { View, Text, TextInput, ScrollView, Platform, Image } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import styles from '../Styles/SearchSuggestionStyles/SearchSuggestion2Style'
import Images from '../../Themes/Images'

type Props = {
  placeholder: string,
  rightEndText: string,
  zIndex: number
}

const SearchSuggestion2 = (props: Props) => {
  const [value, setValue] = useState(false)
  const [searchItem, setSearchItem] = useState('')

  const { placeholder, rightEndText, zIndex } = props

  const changeTextHandler = (text: string) => {
    let val = false
    if (text !== '') {
      val = true
    }
    setValue(val)
    setSearchItem(text)
  }

  const closeDropDown = () => {
    setValue(false)
  }

  return (
    <View style={styles.container}>
      <View style={styles.searchBar}>
        <TextInput
          placeholder={placeholder}
          style={styles.inputField}
          selectionColor="#7B46E4"
          keyboardType={Platform.OS === 'ios' ? 'default' : 'visible-password'}
          onChangeText={changeTextHandler}
          onBlur={closeDropDown}
        />
        <Text style={styles.optionalText}>{rightEndText}</Text>
      </View>

      {value && (
        <View style={styles.dropDownBody} zIndex={zIndex}>
          <ScrollView>
            <TouchableOpacity>
              <View style={styles.firstItem}>
                <Image source={Images.currentLocationIconSmall} style={styles.locationIcon} />
                <Text style={styles.currentLocationText}>Use Current location</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity>
              <View style={styles.listItem}>
                <Text style={styles.listItemText}>{searchItem}</Text>
                <Text style={styles.randomText}>randomtext2</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity>
              <View style={styles.listItem}>
                <Text style={styles.listItemText}>{searchItem}</Text>
                <Text style={styles.randomText}>randomtext3</Text>
              </View>
            </TouchableOpacity>
          </ScrollView>
        </View>
      )}
    </View>
  )
}

SearchSuggestion2.defaultProps = {
  placeholder: 'Search',
  rightEndText: '(Optional)',
  zIndex: 1
}

export default SearchSuggestion2
