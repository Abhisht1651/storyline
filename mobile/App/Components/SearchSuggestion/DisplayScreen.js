import React from 'react'
import { View } from 'react-native'
import SearchSuggestion from './SearchSuggestion'
import SearchSuggestion2 from './SearchSuggestion2'
import styles from '../Styles/SearchSuggestionStyles/DisplayScreenStyle'

const DisplayScreen = () => {
  return (
    <View style={styles.container}>
      <View style={styles.box}>
        <SearchSuggestion />
      </View>
      <View style={styles.box}>
        <SearchSuggestion2 />
      </View>
    </View>
  )
}

export default DisplayScreen
