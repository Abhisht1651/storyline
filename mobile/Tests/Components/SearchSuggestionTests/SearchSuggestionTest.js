import { shallow } from 'enzyme'
import React from 'react'

import SearchSuggestion from '../../../App/Components/SearchSuggestion/SearchSuggestion'

describe('SearchSuggestion', () => {
  it('should match to snapshot', () => {
    const component = shallow(<SearchSuggestion />)
    expect(component).toMatchSnapshot()
  })
})
