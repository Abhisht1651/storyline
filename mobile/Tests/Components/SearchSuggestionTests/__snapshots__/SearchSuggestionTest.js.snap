// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSuggestion should match to snapshot 1`] = `
<View
  style={
    Object {
      "height": 49.285714285714285,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#7B46E4",
        "flexDirection": "row",
        "height": 49.285714285714285,
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Component
      allowFontScaling={true}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Search"
      rejectResponderTermination={true}
      selectionColor="#7B46E4"
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 30,
          "letterSpacing": -0.12,
          "margin": 0,
          "padding": 0,
          "paddingBottom": 9,
          "textAlignVertical": "center",
          "width": "70.72%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Light",
          "fontSize": 23,
          "paddingBottom": 17,
          "textAlignVertical": "center",
        }
      }
    >
      (Optional)
    </Text>
  </View>
</View>
`;
